### DISCUSSION TITLE
Support for postgres range types

### DISCUSSION THREAD

**Discussion by rofinn** [duckdb/duckdb#4459] [General]
Hi there,

First off, thanks for all the hard work on this really cool software! I'm amazed at how fast and simple it is for handling my data wrangling workflows. One thing I've been hacking around is the lack of `tsrange` types and `&&` and `@>` operators. Would just defining a struct alias and some functions / operators be reasonable? I guess something like this should work?

```
CREATE TYPE TSRANGE AS STRUCT(a TIMESTAMP, b TIMESTAMP, bounds UTINYINT)
CREATE FUNCTION &&(a, b) --> Bool
```

I realize without a `GiST` index the operations won't be as performant, but it'd be nice for reducing duplicated SQL code and might mean I could also use `duckdb` for unit testing or existing postgres code :)

**Comment by hawkfish** [comment]
Thanks for the suggestion @rofinn I think your struct approach should get translated into efficient code, but you might have to settle for more verbose language on the intersection operators.

We have a [very efficient range join operator](https://duckdb.org/2022/05/27/iejoin.html) so it might be fast enough for your needs without an index (it basically constructs one at runtime). Let us know how it goes.
